tpm2/tpm2.go

```diff
@@ -985,7 +985,7 @@ func encodeSign(key tpmutil.Handle, password string, digest []byte) ([]byte, err
 	if err != nil {
 		return nil, err
 	}
-	params, err := tpmutil.Pack(digest, AlgNull, tagHashCheck, HandleNull, []byte(nil))
+	params, err := tpmutil.Pack(digest, sigScheme, hashAlg, tagHashCheck, HandleNull, []byte(nil))
```

Comment:
This only works for RSA, HMAC and some ECC signing schemes.
Anonymous ECC schemes also take a count (see section 11.2.1.3 in https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf).

Ideally, switch on `sigScheme`:
- for `AlgNull` do nothing
- for `AlgECDAA` add hash and count
- for the rest add a hash

Reply:
So, sigScheme should be an interface, like crypto.SignerOpts, which implements HashFunc() and , in case of AlgECDAA, also Count() ?

Reply:
If you want to abstract it, make it like https://github.com/google/go-tpm/blob/master/tpm2/structures.go#L72
Have a dedicated type and switch within to decide how to encode.

Or just add a switch here. Up to you.